KanBanApi/KanBanApi.Application/Commands/DeleteTodo/DeleteTodoCommandHandler.cs

```diff
@@ -0,0 +1,26 @@
+using FastEndpoints;
+using KanBanApi.Application.Services;
+using KanBanApi.Domain.Entities;
+using Microsoft.AspNetCore.Identity;
+
+namespace KanBanApi.Application.Commands.DeleteTodo;
+
+public class DeleteTodoCommandHandler(IUnitOfWork _unitOfWork, IActiveSession _activeSession, UserManager<IdentityUser> _userManager) : CommandHandler<DeleteTodoCommand>
+{
+    public override async Task ExecuteAsync(DeleteTodoCommand command, CancellationToken ct = default)
+    {
+        var userId = _activeSession.UserId();
+        if (userId is null) ThrowError("No valid active user");
+
+        var user = await _userManager.FindByIdAsync(userId);
+        if (user is null) ThrowError("No valid active user");
+
+        var todoRepo = _unitOfWork.GetRepository<Todo>();
+
+        var todo = await todoRepo.GetByIdAsync(Guid.Parse(command.TodoId));
+
+        if (todo is null || todo.OwnerId != userId) ThrowError("This todo does not exists");
```

Comment:
This is maybe clever, but to me is a bit ugly. The Handler is taking too much responsibility by verifying the ownership of the todo before proceeding to delete it. We should've created a service (the handlers are just handlers, but not services per se, as per DDD, as far as I recall). So the service of deleting the todo should accept only a todo that is owned by the active user. This is just a comment to exchange ideas, the code is acceptable as it is. 

The same separation of concerns between handlers and sevices applies to the previous handlers i've reviewed so far.